src/main/java/uk/gov/hmcts/dts/fact/services/validation/PostcodeValidator.java

```diff
@@ -58,8 +62,12 @@ public boolean postcodeDataExists(final String postcode) {
                 partialPostcodeValid(PARTIAL_POSTCODE_EDGECASE, postcode, 1));   // if one letters proceeds
     }
 
-    public static boolean isFullPostcodeFormat(final String postcode) {
-        return postcode.matches(FULL_POSTCODE);
+    public boolean fullPostcodeValid(String postcode) {
+        if (StringUtils.isNotBlank(postcode) && isFullPostcodeFormat(postcode)) {
+            return fullPostCodeDataExists(postcode);
+        }
+        log.info("'{}' is not a full postcode", postcode);
```

Comment:
Is this an error scenario? If so maybe log.error is better or log.warn 

If we want to trace the behaviour of the service in live around a users interaction with the service, would the slug being included in the log also help us to narrow down who made the request? Although likely not a problem given the amount of users who will be using this